src/models/injector/Injector.ts

```diff
@@ -0,0 +1,27 @@
+import "reflect-metadata";
+import {Type} from "./GenericClassDecorator";
+
+/**
+ * The Injector stores services and resolves requested instances.
+ */
+export const Injector = new class {
```

Comment:
Nice way to deal with dependency injections without having to add another framework 👍